aws/aws.go

```diff
@@ -69,3 +78,24 @@ func WriteToShell(c *Credentials, windows bool, w io.Writer) {
 		)
 	}
 }
+
+// GetNonExpiredCredentials Return profiles which have a aws_expiration key but are not yet expired
+func GetNonExpiredCredentials(filename string) (Profiles, error) {
+	profiles := Profiles{}
+	cfg, err := ini.LooseLoad(filename)
+	if err != nil {
+		return profiles, err
+	}
+	for _, s := range cfg.Sections() {
+		if s.HasKey("aws_expiration") {
+			v, err := s.Key("aws_expiration").TimeFormat(time.RFC3339)
+			if err == nil {
```

Comment:
The Golang idiom here is to check if we **do** have an error. If we do, we return. If not, we keep going.
This removes one layer of nesting and makes the code more readable. So:

```
if err != nil {
    return nil, err
}

if time.Now()...
```